tests/posix/common/src/signal.c

```diff
@@ -0,0 +1,241 @@
+
+	rc = sigaddset(&set, _NSIG + 1);
+	zassert_equal(rc, -1, "rc should be -1");
+	zassert_equal(errno, EINVAL, "errno should be %s", "EINVAL");
+}
+
+#define SIG_WD_IDX(_signo)  ((int)((_signo - 1) / BITS_PER_LONG))
+#define SIG_MSK_SET(_signo) (WRITE_BIT(sig_mask[SIG_WD_IDX(_signo)], (_signo - 1) % _NSIG_BPW, 1))
+ZTEST(posix_apis, test_signal_addset)
+{
+	int signo;
+	sigset_t set = {0};
+	unsigned long sig_mask[_NSIG_WORDS] = {0};
+
+	signo = SIGHUP;
+	SIG_MSK_SET(signo);
+	zassert_ok(sigaddset(&set, signo));
+	for (int i = 0; i < _NSIG_WORDS; i++) {
+		zassert_equal(set.sig[i], sig_mask[i],
+			      "set.sig[%d of %d] has content: %lx, expected %lx", i, _NSIG_WORDS,
+			      set.sig[i], sig_mask[i]);
+	}
+
+	signo = SIGKILL;
+	SIG_MSK_SET(signo);
+	zassert_ok(sigaddset(&set, signo));
+	for (int i = 0; i < _NSIG_WORDS; i++) {
+		zassert_equal(set.sig[i], sig_mask[i],
+			      "set.sig[%d of %d] has content: %lx, expected %lx", i, _NSIG_WORDS,
+			      set.sig[i], sig_mask[i]);
+	}
+
+	signo = SIGSYS;
+	SIG_MSK_SET(signo);
+	zassert_ok(sigaddset(&set, signo));
+	for (int i = 0; i < _NSIG_WORDS; i++) {
+		zassert_equal(set.sig[i], sig_mask[i],
+			      "set.sig[%d of %d] has content: %lx, expected %lx", i, _NSIG_WORDS,
+			      set.sig[i], sig_mask[i]);
+	}
+
+	signo = SIGRTMIN;
+	SIG_MSK_SET(signo);
+	zassert_ok(sigaddset(&set, signo));
+	for (int i = 0; i < _NSIG_WORDS; i++) {
+		zassert_equal(set.sig[i], sig_mask[i],
+			      "set.sig[%d of %d] has content: %lx, expected %lx", i, _NSIG_WORDS,
+			      set.sig[i], sig_mask[i]);
+	}
+
+	signo = SIGRTMIN + 1;
+	SIG_MSK_SET(signo);
+	zassert_ok(sigaddset(&set, signo));
+	for (int i = 0; i < _NSIG_WORDS; i++) {
+		zassert_equal(set.sig[i], sig_mask[i],
+			      "set.sig[%d of %d] has content: %lx, expected %lx", i, _NSIG_WORDS,
+			      set.sig[i], sig_mask[i]);
+	}
+
+	signo = SIGRTMAX;
+	SIG_MSK_SET(signo);
+	zassert_ok(sigaddset(&set, signo));
+	for (int i = 0; i < _NSIG_WORDS; i++) {
+		zassert_equal(set.sig[i], sig_mask[i],
+			      "set.sig[%d of %d] has content: %lx, expected %lx", i, _NSIG_WORDS,
+			      set.sig[i], sig_mask[i]);
+	}
+}
+
+ZTEST(posix_apis, test_signal_delset_oor)
+{
+	int rc;
+	sigset_t set;
+
+	rc = sigdelset(&set, -1);
+	zassert_equal(rc, -1, "rc should be -1");
+	zassert_equal(errno, EINVAL, "errno should be %s", "EINVAL");
+
+	rc = sigdelset(&set, 0);
+	zassert_equal(rc, -1, "rc should be -1");
+	zassert_equal(errno, EINVAL, "errno should be %s", "EINVAL");
+
+	rc = sigdelset(&set, _NSIG + 1);
+	zassert_equal(rc, -1, "rc should be -1");
+	zassert_equal(errno, EINVAL, "errno should be %s", "EINVAL");
+}
+
+#ifdef CONFIG_64BIT
+ZTEST(posix_apis, test_signal_delset_64)
+{
+	sigset_t set;
+	unsigned long sig_mask = -1;
+
+	memset(&set, -1, sizeof(set));
+
+	WRITE_BIT(sig_mask, SIGHUP - 1, 0);
+	zassert_ok(sigdelset(&set, SIGHUP));
+	zassert_equal(set.sig[0], sig_mask, "set.sig[%d of %d] has content: %lx, expected %lx", 0,
```

Comment:
this test is failing
```
twister -i -p qemu_arc_hs6x   -s tests/posix/common/portability.posix.common.tls
```
https://github.com/zephyrproject-rtos/zephyr/actions/runs/5478149532/jobs/9978281717?pr=60083#step:13:765